site/content/how-to/traffic-management/snippets.md

```diff
@@ -0,0 +1,390 @@
+
+---
+
+## Configure rate limiting to the coffee application
+
+Configure rate limiting to the coffee application by adding the following `SnippetsFilter`:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.nginx.org/v1alpha1
+kind: SnippetsFilter
+metadata:
+  name: coffee-rate-limiting-sf
+spec:
+  snippets:
+    - context: http
+      value: limit_req_zone \$binary_remote_addr zone=coffeezone:10m rate=1r/s;
+    - context: http.server.location
+      value: limit_req zone=coffeezone burst=3 nodelay;
+EOF
+```
+
+The Snippet uses the NGINX `limit_req_module` to configure rate limiting for this HTTPRoute and the
+backend coffee application. This snippet will limit the request processing rate to 1 request per second, and if there
+are more than 3 requests in queue, it will throw a 503 error.
+
+Verify that the `SnippetsFilter` is Accepted:
+
+```shell
+kubectl describe snippetsfilters.gateway.nginx.org coffee-rate-limiting-sf
+```
+
+You should see the following status:
+
+```text
+Status:
+  Controllers:
+    Conditions:
+      Last Transition Time:  2024-10-21T22:20:22Z
+      Message:               SnippetsFilter is accepted
+      Observed Generation:   1
+      Reason:                Accepted
+      Status:                True
+      Type:                  Accepted
+    Controller Name:         gateway.nginx.org/nginx-gateway-controller
+Events:                      <none>
+```
+
+To use the `SnippetsFilter`, update the coffee HTTPRoute to reference it:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: coffee
+spec:
+  parentRefs:
+    - name: gateway
+      sectionName: http
+  hostnames:
+    - "cafe.example.com"
+  rules:
+    - matches:
+        - path:
+            type: PathPrefix
+            value: /coffee
+      filters:
+        - type: ExtensionRef
+          extensionRef:
+            group: gateway.nginx.org
+            kind: SnippetsFilter
+            name: coffee-rate-limiting-sf
+      backendRefs:
+        - name: coffee
+          port: 80
+EOF
+```
+
+Verify that the coffee HTTPRoute's has been configured correctly:
+
+```shell
+kubectl describe httproutes.gateway.networking.k8s.io coffee
+```
+
+You should see the following conditions:
+
+```text
+Conditions:
+      Last Transition Time:  2024-10-28T00:33:08Z
+      Message:               The route is accepted
+      Observed Generation:   2
+      Reason:                Accepted
+      Status:                True
+      Type:                  Accepted
+      Last Transition Time:  2024-10-28T00:33:08Z
+      Message:               All references are resolved
+      Observed Generation:   2
+      Reason:                ResolvedRefs
+      Status:                True
+      Type:                  ResolvedRefs
+```
+
+Test that the `SnippetsFilter` is configured and has successfully applied the rate limiting NGINX configuration changes.
```

Comment:
I think it would be beneficial to describe how the snippets in the referenced SnippetsFilter are injected into the NGINX config. For example, the snippet "limit_req_zone $binary_remote_addr zone=coffeezone:10m rate=1r/s;` directive is injected into the http context and the "limit_req zone=coffeezone burst=3 nodelay;" directive is injected into the location(s) for `/coffee`.

Reply:
See the above coment

Reply:
resolving this thread since I created a new comment elsewhere